include/geos/index/strtree/TemplateSTRtree.h

```diff
@@ -145,6 +146,25 @@ class TemplateSTRtreeImpl {
         nodes.reserve(finalSize);
     }
 
+    /**
+     * Copy constructor, needed because mutex is not copyable
+     */
+    TemplateSTRtreeImpl(const TemplateSTRtreeImpl& other) :
```

Comment:
As mentioned in the top post, I needed to add a copy constructor because the default one doesn't work because the std::mutex lock_ is not copyable, but no idea if this is a sensible implementation.

Reply:
I wouldn't expect a tree to be copied, except possibly as a member variable somewhere else that is initialized later. Another alternative might be to remove the use of the copy constructor.

Reply:
The compilation error pointed to 

https://github.com/libgeos/geos/blob/fbaaba88c18cacd4de4263680c9d071a225590ce/src/algorithm/locate/IndexedPointInAreaLocator.cpp#L57

Reply:
I suppose this copy assignment can be avoided with a different way to construct the index here? (also `TemplateSTRtreeTest.cpp` would need to be updated)